### PR TITLE
[Merged by Bors] - chore(RingTheory/Binomial): make a variable implicit, misc cleanups

### PR DESCRIPTION
**Pull request by ScottCarnahan**
This PR changes the `Nat` variable in `nsmul_right_injective` to implicit.
Also, the `BinomialRing` instance on `Nat` is changed to use `Nat.multichoose`, since (contrary to my previous impression) that function seems unlikely to be deprecated.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Binomial.lean [hunk 1]
```diff
@@ -254,13 +254,14 @@ instance Int.instBinomialRing : BinomialRing ℤ where
         ← Int.neg_ofNat_succ, ascPochhammer_smeval_neg_eq_descPochhammer]
 
 noncomputable instance {R : Type*} [AddCommMonoid R] [Module ℚ≥0 R] [Pow R ℕ] : BinomialRing R where
-  nsmul_right_injective n hn r s hrs := by
+  nsmul_right_injective hn r s hrs := by
+    rename_i n
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Binomial.lean:258]:
```suggestion
  nsmul_right_injective {n} hn r s hrs := by
```

File: Mathlib/RingTheory/Binomial.lean [hunk 2]
```diff
@@ -254,13 +254,14 @@ instance Int.instBinomialRing : BinomialRing ℤ where
...
     rw [← one_smul ℚ≥0 r, ← one_smul ℚ≥0 s, show 1 = (n : ℚ≥0)⁻¹ • (n : ℚ≥0) by simp_all]
     simp_all only [smul_assoc, Nat.cast_smul_eq_nsmul]
   multichoose r n := (n.factorial : ℚ≥0)⁻¹ • Polynomial.smeval (ascPochhammer ℕ n) r
   factorial_nsmul_multichoose r n := by
-    simp only [← smul_assoc]
-    field_simp
+    rw [← smul_assoc, nsmul_eq_mul, mul_inv_cancel₀ (Nat.cast_ne_zero.mpr
+      (Nat.factorial_ne_zero n)), one_smul]
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Binomial.lean:264]: I'd suggest
```suggestion
    rw [← smul_assoc, nsmul_eq_mul, mul_inv_cancel₀ (Nat.cast_ne_zero.mpr n.factorial_ne_zero),
      one_smul]
```

**Review by Ruben-VandeVelde — APPROVED** [review]
This seems reasonable

File: Mathlib/RingTheory/Binomial.lean [hunk 3]
```diff
@@ -254,13 +254,13 @@ instance Int.instBinomialRing : BinomialRing ℤ where
...
 noncomputable instance {R : Type*} [AddCommMonoid R] [Module ℚ≥0 R] [Pow R ℕ] : BinomialRing R where
-  nsmul_right_injective n hn r s hrs := by
+  nsmul_right_injective {n} hn r s hrs := by
     rw [← one_smul ℚ≥0 r, ← one_smul ℚ≥0 s, show 1 = (n : ℚ≥0)⁻¹ • (n : ℚ≥0) by simp_all]
     simp_all only [smul_assoc, Nat.cast_smul_eq_nsmul]
   multichoose r n := (n.factorial : ℚ≥0)⁻¹ • Polynomial.smeval (ascPochhammer ℕ n) r
   factorial_nsmul_multichoose r n := by
-    simp only [← smul_assoc]
-    field_simp
+    rw [← smul_assoc, nsmul_eq_mul, mul_inv_cancel₀ (Nat.cast_ne_zero.mpr n.factorial_ne_zero),
```
Comment (grunweg) [inline Mathlib/RingTheory/Binomial.lean:262]: What's the motivation for this change? Is it faster?
(In general, I believe un-automating is not wanted - unless it results in a big speed-up, or similar.)

**Review by grunweg — COMMENTED** [review]
The implicitness looks good. Some expert please confirm whether the change to the instance is desirable. (And/or comment on the un-golfing.) Thanks for the clean-up!
maintainer delegate?

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/RingTheory/Binomial.lean (hunk 3 above)
Comment (ScottCarnahan) [reply Mathlib/RingTheory/Binomial.lean:262]: Okay, I will undo the change.
